project-koku/koku
# Refreshing materialized views after provider delete

This change will call the materialized view refresh logic when a provider is deleted and when report data expires.

**Testing**
1. Add AWS Source and get database query counts
2. Remove AWS Source
3. Verify that views contain no data

```
postgres=# select count(*) from reporting_aws_compute_summary;
 count 
-------
    62
(1 row)

postgres=# select count(*) from reporting_awscostentrylineitem_daily_summary;
 count 
-------
  1393
(1 row)

koku_server       | [2020-04-01 20:56:40,646] INFO None DELETE: /api/cost-management/v1/sources/f341f191-2166-4a4f-b584-b5fc07c11ba5/ -- ACCOUNT: 10001 USER: user_dev ORG_ADMIN: True REQ_ID: None
koku_server       | [2020-04-01 20:56:41,302] INFO None Provider: AWS Source removed by user_dev
koku_server       | [2020-04-01 20:56:41,423] INFO None Refreshed reporting_aws_compute_summary.
koku_server       | [2020-04-01 20:56:41,509] INFO None Refreshed reporting_aws_compute_summary_by_account.
koku_server       | [2020-04-01 20:56:41,612] INFO None Refreshed reporting_aws_compute_summary_by_region.
koku_server       | [2020-04-01 20:56:41,689] INFO None Refreshed reporting_aws_compute_summary_by_service.
koku_server       | [2020-04-01 20:56:41,760] INFO None Refreshed reporting_aws_cost_summary.
koku_server       | [2020-04-01 20:56:41,833] INFO None Refreshed reporting_aws_cost_summary_by_account.
koku_server       | [2020-04-01 20:56:41,914] INFO None Refreshed reporting_aws_cost_summary_by_region.
koku_server       | [2020-04-01 20:56:41,991] INFO None Refreshed reporting_aws_cost_summary_by_service.
koku_server       | [2020-04-01 20:56:42,061] INFO None Refreshed reporting_aws_database_summary.
koku_server       | [2020-04-01 20:56:42,134] INFO None Refreshed reporting_aws_network_summary.
koku_server       | [2020-04-01 20:56:42,215] INFO None Refreshed reporting_aws_storage_summary.
koku_server       | [2020-04-01 20:56:42,296] INFO None Refreshed reporting_aws_storage_summary_by_account.
koku_server       | [2020-04-01 20:56:42,388] INFO None Refreshed reporting_aws_storage_summary_by_region.
koku_server       | [2020-04-01 20:56:42,468] INFO None Refreshed reporting_aws_storage_summary_by_service.
koku_server       | [2020-04-01 20:56:43,500] INFO None DELETE: /api/cost-management/v1/sources/f341f191-2166-4a4f-b584-b5fc07c11ba5/ 200 -- ACCOUNT: 10001 USER: user_dev ORG_ADMIN: True REQ_ID: None
koku_server       | [2020-04-01 20:56:43,502] INFO None "DELETE /api/cost-management/v1/sources/f341f191-2166-4a4f-b584-b5fc07c11ba5/ HTTP/1.1" 200 9956


postgres=# select count(*) from reporting_awscostentrylineitem_daily_summary;
 count 
-------
     0
(1 row)

postgres=# select count(*) from reporting_aws_compute_summary;
 count 
-------
     0
(1 row)

postgres=# 
```
